front/lib/resources/agent_message_feedback_resource.ts

```diff
@@ -43,130 +61,357 @@ export class AgentMessageFeedbackResource extends BaseResource<AgentMessageFeedb
+        // These clauses are optional
+        ...(agentConfiguration
+          ? {
+              agentConfigurationId: agentConfiguration.sId.toString(),
+            }
+          : {}),
+        ...createdAtClause,
       },
+
+      include: [
+        {
+          model: AgentMessageModel,
+          attributes: ["id"],
+          as: "agentMessage",
+          include: [
+            {
+              model: Message,
+              as: "message",
+              attributes: ["id", "sId"],
+              include: [
+                {
+                  model: Conversation,
+                  as: "conversation",
+                  attributes: ["id", "sId"],
+                },
+              ],
+            },
+          ],
+        },
+        {
+          model: UserResource.model,
+          attributes: ["name", "imageUrl", "email"],
+        },
+      ],
+      order: [
+        ["agentConfigurationVersion", "DESC"],
+        ["createdAt", "DESC"],
+      ],
+      limit: filters?.limit,
     });
 
-    if (!agentMessageFeedback) {
-      return null;
-    }
+    return (
+      agentMessageFeedback
+        // Typeguard needed because of TypeScript limitations
+        .filter(
+          (
+            feedback
+          ): feedback is AgentMessageFeedback & {
+            agentMessage: { message: Message & { conversation: Conversation } };
+          } => !!feedback.agentMessage?.message?.conversation
+        )
+        .map((feedback) => {
+          return {
+            id: feedback.id,
+            messageId: feedback.agentMessage.message.sId,
+            agentMessageId: feedback.agentMessageId,
+            userId: feedback.userId,
+            thumbDirection: feedback.thumbDirection,
+            content: feedback.content
+              ? feedback.content.replace(/\r?\n/g, "\\n")
+              : null,
+            isConversationShared: feedback.isConversationShared,
+            createdAt: feedback.createdAt,
+            agentConfigurationId: feedback.agentConfigurationId,
+            agentConfigurationVersion: feedback.agentConfigurationVersion,
 
-    return new AgentMessageFeedbackResource(
-      AgentMessageFeedback,
-      agentMessageFeedback.get()
+            ...(withMetadata && {
+              // This field is sensitive, it allows accessing the conversation
+              conversationId: feedback.isConversationShared
+                ? feedback.agentMessage.message.conversation.sId
+                : null,
+              userName: feedback.user.name,
+              userEmail: feedback.user.email,
+              userImageUrl: feedback.user.imageUrl,
+            }),
+          };
+        })
     );
   }
 
-  static async fetchByUserAndAgentMessages(
-    user: UserType,
-    agentMessages: AgentMessage[]
-  ): Promise<AgentMessageFeedbackResource[]> {
-    const agentMessageFeedback = await AgentMessageFeedback.findAll({
+  static async getConversationFeedbacksForUser(
```

Comment:
The name is off as this is  for user and conversation right?

Reply:
I guess e.g. `getFeedbacksLeftByUserInConversation` would be more descriptive, but not shockingly misleading in its current state imo

Reply:
Actually nvm the name has Conversation in it. Name LGTM :+1: